Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,196 @@
+as listed on the [community section](https://www.keycloak.org/community)
+of the project website. It uses [Github issues](https://github.com/keycloak/keycloak/issues)
+to track development.
+
+It offers a REST API for administration and there's a separately maintained
+3rd party python module as well as ansible support for it. Both of these are
+downstream of Keycloak itself and may thus not always be feature complete and
+suffer latency with respect to getting adjusted to upstream changes.
+
+It offers support for commonly used SSO protocols and is "reasonably" fast
+in adopting to protocol standard changes and extensions. This has been
+observed in the case of logout support (backend and frontend variants) in OIDC.
+
+It offers a concept of "Identity Brokering", where Keycloak is not just IdP
+but also "client" to other IdPs. This allows daisy chaining of identity
+federation. In this configuration it can work as a point of protocol
+transition between different supported SSO protocols (SAML, OAuth 2.0, etc.).
+
+Beyond this capability of using other IdPs as identity sources, it also supports
+using classic LDAP based IAM services as backend (OpenLDAP and Active Directory,
+e.g.).
+
+Keycloak's implementation makes some design decisions, that are specific
+to it and have consequences for clients of the service. E.g. Keycloak
+has a concept of management "Realms", which have their own specific
+set of HTTP API entrypoints, both for administration as well as for IdP
+requests.
+
+Commonly Keycloak realms can be used to map them 1:1 to user domains,
+but since Keycloak supports configuring multiple backend IdPs in a
+realm to be used for "Identity Brokering", there is always the
+possibility to create a kind of "proxy" realm to provide a single
+standard set of HTTP API endpoints for SSO clients (service providers)
+to avoid the need to frequently extend/reduce client service configuration
+whenever a new IdP federation needs to be added to Keycloak to onboard
+a new customer. This is relevant for services like OpenStack Keystone,
+which currently cannot be easily reconfigured for new SSO endpoints
+without restarting the service, making the service unavailable for
+a short span of time and increasing risk connected with service restarts.
+
+Since version 17, Keycloak claims that it's capability for
+"cloud native" deployments on Kubernetes has improved.
+
+[Keycloak is offering a REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html)
+for all aspects of its administration interface.
+
+For storage of Keycloak configuration and local user metadata
+(e.g. from which external IdP a user account originally came from)
+Keycloak supports several SQL backends through JDBC. Thus
+it can be hooked up to a Postgres Database or to a
+MariaDB/Galera cluster e.g..
+
+### Zitadel
+
+Zitadel is a newer implementation of a SSO IdP. It is implemented
+in Go and under active development.
+
```

Comment:
```suggestion
Zitadel is a newer implementation of a SSO IdP.
It is implemented in Go and under active development and maintained by ZITADEL.

The project is open for community [contributions](https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md)  to all parts of the eco system.
Feature requests and bugs being tracked on [Github](https://github.com/orgs/zitadel/projects/2/views/5) for development.
Community questions can be asked in the [public chat](https://zitadel.com/chat) or via [Github Discussions](https://github.com/zitadel/zitadel/discussions). 

ZITADEL offers support for the commonly used authentication and authorization protocols such as OIDC, OAuth2, SAML2. 
It is a compliant and certified OpenID Connect provider with support for various Grant Types for both human users and machine users.

Compared to Keycloak SPIs, ZITADEL offers Actions to customize and integrate (eg, calling external APIs, Webhooks, customizing pre-built workflows, customizing tokens)
Actions are executed at runtime and can be maintained independently of platform.

Identity brokering (OIDC, SAML, JWT) can be configured system-wide or for each organization with templates.
Users will be created just in time for audit purposes and linked to the external identity provider.
Users can have multiple identity providers linked to their profile.
```

Reply:
Just wanted to add a bit more substance to this section compared to keycloak. Feel free to adjust if you think it is too opinionated.

Reply:
Thanks for the input. I've picked it up.